core/types/rlp_backwards_compat.libevm_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+package types_test
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	. "github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/libevm/ethtest"
+	"github.com/ava-labs/libevm/rlp"
+)
+
+func TestHeaderRLPBackwardsCompatibility(t *testing.T) {
+	// This is a deliberate change-detector test that locks in backwards
+	// compatibility of RLP encoding.
+	rng := ethtest.NewPseudoRand(42)
+
+	const numExtraBytes = 16
+	hdr := &Header{
+		ParentHash:  rng.Hash(),
+		UncleHash:   rng.Hash(),
+		Coinbase:    rng.Address(),
+		Root:        rng.Hash(),
+		TxHash:      rng.Hash(),
+		ReceiptHash: rng.Hash(),
+		// Bloom populated below
+		Difficulty: rng.Uint256().ToBig(),
+		Number:     rng.BigUint64(),
+		GasLimit:   rng.Uint64(),
+		GasUsed:    rng.Uint64(),
+		Time:       rng.Uint64(),
+		Extra:      make([]byte, numExtraBytes), // populated below
+		MixDigest:  rng.Hash(),
+		// Nonce populated below
+
+		BaseFee:          rng.BigUint64(),
+		WithdrawalsHash:  rng.HashPtr(),
+		BlobGasUsed:      rng.Uint64Ptr(),
+		ExcessBlobGas:    rng.Uint64Ptr(),
+		ParentBeaconRoot: rng.HashPtr(),
+	}
+	require.Equal(t, BloomByteLength, rng.Read(hdr.Bloom[:]))
+	require.Equal(t, len(BlockNonce{}), rng.Read(hdr.Nonce[:]))
+	require.Equal(t, numExtraBytes, rng.Read(hdr.Extra))
```

Comment:
That check is rather unneeded, since we test the test really, especially given numExtraBytes is defined in the test and we create the slice inline in this test.

Reply:
Now obsolete since inlining of the random generation.